src/serializers/infer.rs

```diff
@@ -609,16 +609,11 @@ pub(crate) fn infer_json_key_known<'a>(
         }
         ObType::Decimal => Ok(Cow::Owned(key.to_string())),
         ObType::Bool => super::type_serializers::simple::bool_json_key(key),
-        ObType::Str | ObType::StrSubclass => {
-            let py_str = key.downcast::<PyString>()?;
-            Ok(Cow::Owned(py_str.to_str()?.to_string()))
-        }
```

Comment:
While most of this diff is within noise of allocations elsewhere in the system, removing this allocation does show a teeny tiny improvement here:

<img width="890" alt="image" src="https://github.com/user-attachments/assets/13290a21-a0ae-4685-b02b-73801e524913">